### DISCUSSION TITLE
InlineCssTextArea.getTotalWidthEstimate()

### DISCUSSION THREAD

**Discussion by williamclocksin** [FXMisc/RichTextFX#1201] [Q&A] [answered]
Hi Jugen, I have added the code you suggested to bind to the area, and then tried calling area.getTotalWidthEstimate(), but get the same result.  Not sure what to do; I don't wish to dive into the InlineCssTextArea code.  There must be a way to get the width and height of actual text as it would be rendered (including font and other style changes), so I can then adjust the boundaries of the area and its enclosing Box to display it. Do I need to applyCss() or something similar before getTotalWidthEstimate()? The attached screenshot shows the area with the text 'text is too wide to fit in the box' but only the first 13 characters are shown. I would like the orange outline to grow or shrink according to what it in the area.
![image](https://github.com/FXMisc/RichTextFX/assets/143699911/8cfd9a89-2652-4456-b687-84657c9eb137)

**Comment by williamclocksin** [comment]
PS:  I tried area.applyCss() before area.getTotalWidthEstimate(), but got the same wrong result.

**Comment by williamclocksin** [comment, marked as answer]
PS:  Looking at your code, I simply did area.getCharacterBoundsOnScreen(0, area.getLength()) when I update the contents of the area, and I now get correct changes in bounds when I append or delete characters.  I will forget about the binding, and just do this.  Thanks, Jugen, for getting me started on fixing this!

**Comment by Jugen** [comment]
Ahh, sorry I didn't mention that the code that I provided would set the area/field width automatically as the contents changes - so no need for calling area.getTotalWidthEstimate(). Anyway glad you came right. 

Just for other viewers reference here's the code I initially suggested:
```java
var tf = new InlineCssTextField( "Test" );

tf.maxWidthProperty().bind( Bindings.createDoubleBinding( () ->
{
    var charBounds = tf.getCharacterBoundsOnScreen( 0, tf.getLength() );
    if ( charBounds.isEmpty() ) return tf.getPrefWidth();

    var pad = tf.getInsets();
    var width = charBounds.get().getWidth();

    return pad.getLeft() + width + pad.getRight();
},
tf.lengthProperty(), tf.insetsProperty() ));
```

**Reply by williamclocksin** [reply]
Thanks. Yes. Being a Programmer from the Stone Age, I am not as familiar with the Bean philosophy of bindings and properties as I would like to be. But, in my application with a legacy MVC design, I need to have control over how and when structures are changed so the correct graphics can happen on the Canvas. When it is fully working, then maybe I can look at a more modern design.
